src/test/concurrency/stress_test.cpp

```diff
@@ -139,21 +133,77 @@ TEST_F(StressTest, TestTransactionInserts) {
   {
     auto pipeline = SQLPipelineBuilder{std::string{"SELECT MIN(b) FROM table_b"}}.create_pipeline();
-    const auto [_, table] = pipeline.get_result_table();
-    EXPECT_FLOAT_EQ(table->get_value<int32_t>(ColumnID{0}, 0), iterations_per_thread);
+    const auto [_, verification_table] = pipeline.get_result_table();
+    EXPECT_EQ(*verification_table->get_value<int32_t>(ColumnID{0}, 0), iterations_per_thread);
   }
+}
 
-  // Really pessimistic, but at least 2 statements should have made it
-  EXPECT_EQ(successful_increments, num_threads * iterations_per_thread);
-  EXPECT_EQ(conflicted_increments, 0);
+TEST_F(StressTest, TestTransactionInsertsPackedNullValues) {
+  // As ValueSegments store their null flags in a vector<bool>, which is not safe to be modified concurrently,
+  // conflicts may (and have) occurred when that vector was written without any type of protection.
+
+  TableColumnDefinitions column_definitions;
+  column_definitions.emplace_back("a", DataType::Int, false);
+  column_definitions.emplace_back("b", DataType::Int, true);
+  const auto table = std::make_shared<Table>(column_definitions, TableType::Data, Chunk::DEFAULT_SIZE, UseMvcc::Yes);
+  Hyrise::get().storage_manager.add_table("table_c", table);
+
+  const auto iterations_per_thread = 200;
+
+  // Define the work package - each job writes a=job_id, b=(NULL or 1, depending on job_id)
+  std::atomic_int job_id{0};
+  const auto run = [&]() {
+    const auto my_job_id = job_id++;
+    for (auto iteration = 0; iteration < iterations_per_thread; ++iteration) {
+      // b is set to NULL by half of the jobs.
+      auto pipeline = SQLPipelineBuilder{std::string{"INSERT INTO table_c (a, b) VALUES ("} +
+                                         std::to_string(my_job_id) + ", " + (my_job_id % 2 ? "NULL" : "1") + ")"}
+                          .create_pipeline();
+      const auto [status, _] = pipeline.get_result_table();
+      EXPECT_EQ(status, SQLPipelineStatus::Success);
+    }
+  };
+
+  // Create the async objects and spawn them asynchronously (i.e., as their own threads)
+  const auto num_threads = 20u;
+  std::vector<std::future<void>> thread_futures;
+  thread_futures.reserve(num_threads);
+
+  for (auto thread_num = 0u; thread_num < num_threads; ++thread_num) {
+    // We want a future to the thread running, so we can kill it after a future.wait(timeout) or the test would freeze
+    thread_futures.emplace_back(std::async(std::launch::async, run));
+  }
+
+  // Wait for completion or timeout (should not occur)
+  for (auto& thread_future : thread_futures) {
+    // We give this a lot of time, not because we need that long for 100 threads to finish, but because sanitizers and
+    // other tools like valgrind sometimes bring a high overhead that exceeds 10 seconds.
+    if (thread_future.wait_for(std::chrono::seconds(600)) == std::future_status::timeout) {
+      ASSERT_TRUE(false) << "At least one thread got stuck and did not commit.";
```

Comment:
Haven't seen this before. Why can't we use `Fail()`? Does `Fail()` crash the binary and the assert does not?

Reply:
Not sure, didn't write it. Your thought makes sense, it's probably because of that.Generally, we don't seem to use our own assertion in the tests, anyway.